health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/ExerciseSampleApp.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.exercisesamplecompose.presentation
+
+import android.content.ContentValues.TAG
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+import androidx.wear.compose.navigation.SwipeDismissableNavHost
+import androidx.wear.compose.navigation.composable
+import com.example.exercisesamplecompose.Screens
+import com.example.exercisesamplecompose.data.HealthServicesManager
+import kotlinx.coroutines.launch
+
+/** Navigation for the exercise app. **/
+
+@Composable
+fun ExerciseSampleApp(
+    healthServicesManager: HealthServicesManager,
+    navController: NavHostController,
+    startDestination: String
+) {
+    val composableScope = rememberCoroutineScope()
+
+    SwipeDismissableNavHost(
+        navController = navController, startDestination = startDestination
+    ) {
+        composable(Screens.StartingUp.route) {
+            val viewModel: ExerciseViewModel = viewModel(
+                factory = ExerciseViewModelFactory(
+                    healthServicesManager = healthServicesManager, LocalContext.current
+                )
+            )
```

Comment:
If you're using Hilt for injection, you shouldn't need a factory here, instead use val viewModel = hiltViewModel<ExerciseViewModel>()

https://proandroiddev.com/jetpack-compose-adding-a-hilt-viewmodel-to-navigation-28cdc95e28bb

Reply:
Magic!